### ISSUE TITLE
Response from API isn't actually data, it just says 'SUCCESS'

### ISSUE THREAD

**Issue by michaelcuneo** [aws-amplify/amplify-js#773]
Shouldn't I have some data as well?

{"success":"get call succeed!","url":"/ImageSets"}

**Comment by elorzafe** [comment]
Hi @michaelcuneo 

I think you are using awsmobile-cli to create the api. That API has sample code on it, you can go to ```your-project/awsmobilejs/backend/cloud-api/``` there you will find a folder with the API name that contains the source code of your api.

Take a look on ```app.js``` file, there you will see the data that is on response is the same as you mention.

**Comment by michaelcuneo** [comment]
Well, I'm kind of baffled now. They explain how to write a program in there. I don't want to write a program, to run a lambda function, I just need data. Do I need to add my own code in there and return data or something? I thought awsmobile set up the lambda stuff when I created the CRUD

From what I can see here... the example get method doesn't actually get anything.

**Comment by michaelcuneo** [comment]
Yeah, I did all of that, I have a cloud-api and I created a CRUD API for an existing Amazon DynamoDB Table. So I've got create/read/update/delete, I can test all those and the reply comes back as success, every time. There's just no sample of how you actually grab table data from the commands you run against the API. It's all explanation of Cloud Logic.

**Comment by elorzafe** [comment]
What version of awsmobile-cli are you using? Which OS? Can you show me a code snippet? I just tried out and it works as expected.

**Comment by michaelcuneo** [comment]
Seems as though, it doesn't matter what I change the URL to... it always comes back with Success, but empty data.

**Comment by elorzafe** [comment]
Did you solve this issue #307 ?

**Comment by elorzafe** [comment]
Latest awsmobile-cli is 1.1.4. You can check your cli version with this command.
```
awsmobile -V
```

**Comment by michaelcuneo** [comment]
Looks like I have 1.0.21

**Comment by michaelcuneo** [comment]
I think I have found the problem... 

If I go to the Mobile Hub, click on my project, get to the Backend area, with the list of everything I have active and enabled in the App. If I click on Cloud Logic, it shows me my ImageSetsCRUD, Type API, Description, Test API, Actions. X etc...  if I click on Edit the API, I get Updating an API, that looks all normal, protected to all my App's users, lambda function name is right, CORS enabled, etc... All good.

If I click on Edit your function in AWS Lambda, I get to the Lambda -> Functions -> ImageSetsCRUD-ImageSets-mobilehub-[appid] ... and this is the area where it doesn't look good.

There's no Triggers. awsmobile-cli isn't creating triggers for the Lambda functions from the API Gateway. The API Gateway is there, with a configuration suitable to trigger the Lambda, but it's not being created when running the awsmobile cli... 

<img width="1606" alt="shot_from_lambda" src="https://user-images.githubusercontent.com/12700009/40090636-5b6b981c-58f6-11e8-96a1-b828aeb911d7.png">

Now I think if I try and edit this manually, it will say 'Nope this is controlled by the CloudFormation Stack' ... so if I awsmobile pull after editing and manually placing the triggers into the list from the API Gateway, I'll get 'This Lambda function was modified outside of it's CloudFormation stack' errors. And then I'll get stuck in a revert loop because it can't be fixed.

**Comment by michaelcuneo** [comment]
Yes, just as I thought. I modified the Lambda Function and created the trigger that it needed... and now I am unable to access any of it.

'The stack \'arn:aws:cloudformation:ap-southeast-2:885973830331:stack/MOBILEHUB-1347566052-Development/9f795c30-4daa-11e8-a552-50fae94facae\' in Amazon CloudFormation has been modified outside of the Mobile Hub. This stage \'Development\' can no longer be modified inside of Mobile Hub.',

So now I can't run with awsmobile run, until I force the CloudFormation to revert back to the broken state.

**Comment by michaelcuneo** [comment]
When I go into Cloud Logic now, within my project I also get this message... 

Mobile Hub has disabled editing of the Cloud Logic APIs in this project to avoid overwriting advanced settings changes made in the console of another AWS Service. Learn why this has happened.
I understand that further additions, modifications and deletions of features in this Mobile Hub project may result in overwriting configuration changes that were made in other service consoles (i.e. Amazon API Gateway, AWS CloudFormation, AWS Lambda). I accept this risk and I would like such safeguards to be removed.

So I hit 'Unlock Project' but it never completes. The Cloud Formation revert takes over 48 hours and never does a thing... I end up having to delete the entire project and starting it all from scratch.

**Comment by michaelcuneo** [comment]
Just like I thought, I added the trigger because it appears to be essential to me... and now I have errors and no way of fixing it without a complete rebuild of the entire awsmobile project from scratch.

**Comment by elorzafe** [comment]
@michaelcuneo when you modify your api outside of the cli or MobileHub console the feature is locked (is not sync with what is configured outside of MobileHub. That is why you saw that message on MobileHub console. If you disable the feature, call push and then enable again and push that should create a new api. With the correct code. I suggest to not modify api gateway or lambda outside of the cli to avoid that problem.

Please let me know how it goes

**Comment by michaelcuneo** [comment]
Yeah, I've read about it becoming locked before and how to get past that point, but unlocking it was unsuccessful for me, so I restarted the project from scratch, and now I'm at the point of recreating the API's. I did a complete test project and noticed that when the Lambda's are created there are no triggers no matter if I do it from scratch properly, or not. So I guess the Lambda's are triggered in some other manner that doesn't require and actual trigger in the Console GUI.

I've been working on the project faster than people are able to reply to me, so by the time I have new solutions I have more problems. But... I have double checked over and over, by creating multiple sample projects with different configurations... and no Lambda Triggers are ever made by the awsmobile CLI. Wether or not that's a normal thing to occur... no idea. But it still won't allow API calls from scratch. Do I have an issue with my entire AWS Account or something?

**Comment by hollyewhite** [comment]
@michaelcuneo - Did you ever find any helpful information on this or resolve it? I'm struggling with a very similar issue.

**Comment by elorzafe** [comment]
@hollygirouard did you solve your issue, what were you struggling with?

**Comment by michaelcuneo** [comment]
@hollygirouard I need to look up my documents to find out what I did to fix this, but I remember that I know I did. The old generation of API Rest code was not actually generating code based on listing/getting properly. The listing and getting were not actually being performed, the command was being called successfully but the command was to do literally ... nothing.

I had to add this method to backend/cloud-api/imageSets because the existing database enabled, didn't provide a method to list everything.

```
app.get('/imageSets', function(req, res) {
  var params = {
    TableName: tableName,
    Select: 'ALL_ATTRIBUTES',
  };
  dynamodb.scan(params, (err, data) => {
    if (err) {
      res.json({error: 'Could not load items: ' + err.message});
    }
    res.json({
      data: data.Items.map(item => {
        return item;
      })
    });
  });
});
```

I have switched all my future projects from this Rest method to GraphQL, which is highly recommended... but for anyone still hammering away on this issue, that's the solution.
